nodejs/extensions.bzl

```diff
@@ -14,54 +26,65 @@ def _toolchain_extension(module_ctx):
                     # Prioritize the root-most registration of the default node toolchain version and
                     # ignore any further registrations (modules are processed breadth-first)
                     continue
-                if toolchain.node_version == registrations[toolchain.name].node_version and toolchain.node_version_from_nvmrc == registrations[toolchain.name].node_version_from_nvmrc:
+                toolchain_info = _toolchain_info(toolchain)
+                registered_toolchain_info = _toolchain_info(registrations[toolchain.name])
+                if toolchain_info != registered_toolchain_info:
```

Comment:
Lets change this to `if not toolchains_equal(toolchain, registrations[toolchain.name])` just to hide most of this stuff (the `_toolchain_info` calls) in one method that describes what's going on.